modules/extension/mbstyle/src/main/java/org/geotools/mbstyle/parse/MBObjectParser.java

```diff
@@ -332,9 +332,10 @@ public String get(JSONArray json, int index) {
         }
         if (index < json.size() && json.get(index) instanceof String) {
             return (String) json.get(index);
-        }
-        if (index < json.size() && json.get(index) instanceof JSONArray) {
-            return MBExpression.transformExpression((JSONArray) json.get(index)).toString();
+            //        }
+            //        if (index < json.size() && json.get(index) instanceof JSONArray) {
+            //            return MBExpression.transformExpression((JSONArray)
+            // json.get(index)).toString();
```

Comment:
This blind conversion to a string incorrectly resulted in correctly generated Expressions being represented as strings. I suspect it should be removed!

Reply:
I agree, this looks like a good change.